community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/ReadAheadChannel.java

```diff
@@ -56,7 +56,7 @@ public ReadAheadChannel( T channel, int readAheadSize )
 
     public long position() throws IOException
     {
-        return channel.position();
+        return offset();
```

Comment:
@martinfurmanski I'm not sure I completely understand this change and it actually looks a bit strange. Couple questions:
 1) we read the underlying file forward and buffer some data, so `position` actually means amount of bytes read; that is why we do not count the buffered read-ahead bytes; is this correct?
 2) if `#position()` simply delegates to `#offset()` than there should be only one of them present; i'd vote for `#position()`
 3) it would be really nice to have a small javadoc saying that it is simply amount of bytes between position 0 and current position
 4) there should definitely be a test for this change because it looks like a bugfix and we can't let it regress in future again